lib/schema/schema.rb

```diff
@@ -14,10 +16,20 @@ def attribute_macro(attr_name, type=nil, strict: nil, default: nil)
         raise Schema::Attribute::Error, "The \"#{attr_name}\" attribute is declared with the \"strict\" option but a type is not specified"
       end
 
+      if type == Boolean && strict == false
+        raise Schema::Attribute::Error, "The \"#{attr_name}\" attribute is declared with the \"strict\" option disabled but boolean type is specified"
+      end
+
```

Comment:
This particular bit of defensive coding seemed necessary, for a rather esoteric and confusing reason. Bear with me as I try to explain.

Suppose a user decided they wanted a non-strict boolean attribute. This would imply they plan on passing in objects that belonged to _subclasses_ of TrueClass or FalseClass. However, TrueClass and FalseClass cannot actually be subclassed (not even with hacky workarounds like overriding `.new`). Therefore, the user could never actually accomplish the behavior they intended by deliberately setting `strict` to false.

Part of me wants to take this check out, simply because the user would be trying to do something rather nonsensical. I can't think of a sound reason for _wanting_ to subclass TrueClass or FalseClass, and even if users tried, they couldn't. I added this check, though, because our interface allows the particular combination of Boolean as the type and strict set to false, and it seemed important to have all bases covered.